### DISCUSSION TITLE
JPEG Encoder for AIR application

### DISCUSSION THREAD

**Discussion by zabdoul60460** [airsdk/Adobe-Runtime-Support#2414] [Q&A]
Hi,
Before working with Harman AIR (33.1.1...), I used JPGAlchemyEncoder to encode BitmapData to JPEG.
It works fast (around 2s to encode).
Now with Harman AIR the same BitmapData takes around 1m to encode the same BitmapData on Mac OS !
Could you help me to find  a better solution ?
Best regards

**Comment by FischMich** [comment]
There's a native "encode" option for BitmapData since a while.
It's very fast!

Docs:
https://airsdk.dev/reference/actionscript/3.0/

 

// Compress a BitmapData object as a JPEG file.

     var bitmapData:BitmapData = new BitmapData(640,480,false,0x00FF00);

     var byteArray:ByteArray = new ByteArray();

     bitmapData.encode(new Rectangle(0,0,640,480), new flash.display.JPEGEncoderOptions(), byteArray);

**Reply by ajwfrost** [reply]
I would hope that this native `BitmapData.encode()` method is at least as fast as an Alchemy-based approach .. but @zabdoul60460 are you able to confirm what API you are using when you say it takes around a minute? Plus what macOS machine have you got and what mode is the application running in? (I am wondering if a potential use of Rosetta2 is slowing it down a bit..)

thanks

**Comment by zabdoul60460** [comment]
Hello,

After updating Adobe Air runtime (from 33.1.1.731 to 50.1) encoding 
process is fast !

Best regards


      IsabelleLOYER

InterSystèmes

	

	
	

	

	+ 33 1 34 69 22 66 <tel:+ 33 1 34 69 22 66> | +33 6 75 72 82 93 
<tel:+33 6 75 72 82 93>

	***@***.*** ***@***.***>

	www.intersystemes.fr <//www.intersystemes.fr>

	20 rue de Montubois, 95840 Béthemont la Forêt



Le 12/01/2023 à 14:43, Andrew Frost a écrit :
